contracts/protocol/clients/FermionFNFT.sol

```diff
@@ -72,38 +74,15 @@ contract FermionFNFT is FermionFractions, FermionWrapper, ERC2771Context, IFermi
         initializeWrapper(_owner, _metadataUri);
         intializeFractions(_exchangeToken);
 
-        string memory _offerIdString = Strings.toString(_offerId);
-        __ERC721_init(string.concat("Fermion FNFT ", _offerIdString), string.concat("FFNFT_", _offerIdString));
-    }
-
-    /**
-     * @notice Updates the name of the token
-     * @dev Only callable by the contract owner
-     * @param _name The new name for the token
-     */
-    function setName(string memory _name) external onlyOwner {
-        Common._getERC721Storage()._name = _name;
-    }
+        if (bytes(_tokenMetadata.name).length == 0) {
+            _tokenMetadata.name = "Fermion FNFT";
 
-    /**
-     * @notice Updates the symbol of the token
-     * @dev Only callable by the contract owner
-     * @param _symbol The new symbol for the token
-     */
-    function setSymbol(string memory _symbol) external onlyOwner {
-        Common._getERC721Storage()._symbol = _symbol;
-    }
+            string memory _offerIdString = Strings.toString(_offerId);
+            _tokenMetadata.name = string.concat("Fermion FNFT ", _offerIdString);
+            _tokenMetadata.symbol = string.concat("FFNFT_", _offerIdString);
+        }
```

Comment:
```suggestion
        }

        if (bytes(_tokenMetadata.symbol).length == 0) {
            _tokenMetadata.symbol = string.concat("FFNFT_", _offerIdString);
        }
```
I guess it makes more sense to set default value for symbol if not set in the arguments

Reply:
That makes sense, otherwise they can submit a empty symbol.

I just implemented it a bit differently, since your suggestion requires to either construct `_offerIdString` twice are always construct it before even checking if any field is empty.  

Just doing a few more if statement is a bit more efficient.